File: Dia3/Ruby/Seguro/CuentaSeguro.rb
class CuentaSeguro
    attr_reader :numeroPoliza, :importe

    def numeroPoliza=(numeroPoliza)
        if numeroPoliza.strip.empty?
            raise ArgumentError, "El nombre de la poliza no puede estar vacio"    
        end

        @numeroPoliza = numeroPoliza

    end

    def importe=(importe)
        if importe < 0
            raise ArgumentError, "El importe de la poliza no puede ser negativo"    
        end

        @importe = importe
    end

    def initialize(numeroPoliza, importe)
        self.numeroPoliza = numeroPoliza
        self.importe = importe
    end

    def to_s()
        "Poliza nº: #{numeroPoliza}, Importe: #{importe}"
    end

end

File: Dia3/Java/Seguro/src/CuentaSeguro.java
public class CuentaSeguro {
    private String numPoliza;
    private double importe;

    public CuentaSeguro(String numPoliza, double importe) {
        setNumPoliza(numPoliza);
        setImporte(importe);
    }

    public String getNumPoliza() {
        return numPoliza;
    }

    public void setNumPoliza(String numPoliza) {

        if (numPoliza == null || numPoliza.isEmpty()) {
            throw new IllegalArgumentException("El numero de la poliza no puede estar vacio");
        }

        this.numPoliza = numPoliza;
    }

    public double getImporte() {
        return importe;
    }

    public void setImporte(double importe) {

        if (importe < 0) {
            throw new IllegalArgumentException("El importe no puede ser negativo");
        }

        this.importe = importe;
    }

    @Override
    public String toString() {
        return String.format("Póliza nº: %s, Importe: %.2f", numPoliza, importe);
    }

}
